torchrl/collectors/collectors.py

```diff
@@ -786,38 +784,14 @@ def iterator(self) -> Iterator[TensorDictBase]:
                     # >>> assert data0["done"] is not data1["done"]
                     yield tensordict_out.clone()
 
-    def _step_and_maybe_reset(self) -> None:
-
-        self._tensordict = step_mdp(
-            self._tensordict,
-            reward_keys=self.env.reward_keys,
-            done_keys=self.env.done_keys,
-            action_keys=self.env.action_keys,
-        )
-        if not self.reset_when_done:
-            return
-        td_reset = self._tensordict.clone(False)
-        any_done = terminated_or_truncated(
-            td_reset,
-            full_done_spec=self.env.output_spec["full_done_spec"],
-            key="_reset",
+    def _update_traj_ids(self, tensordict) -> None:
+        # we can't use the reset keys because they're gone
+        traj_sop = _aggregate_resets(
+            tensordict.get("next"), done_keys=self.env.done_keys
```

Comment:
I assume we have to do this becuse there are no reset keys anymore visible by the collector.
Since this is quite counterintitive, what about a `_aggregate_dones()`?
I also have other suggestions in the other comment relating to this function.

Reply:
yep we can rename the function. I would rather say that it's more direct to aggregate the done rather than the "_reset" which come from the dones